pallets/crowdloan-rewards/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+		type RelayChainAccountId: Parameter + Member + MaybeSerializeDeserialize + Default;
+
+		/// The total vesting period.
+		type VestingPeriod: Get<Self::BlockNumber>;
+	}
+
+	type BalanceOf<T> = <<T as Config>::RewardCurrency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+	/// Stores info about the rewards owed as well as how much has been vested so far.
+	/// For a primer on this kind of design, see the recipe on compounding interest
+	/// https://substrate.dev/recipes/fixed-point.html#continuously-compounding
+	#[derive(Default, Clone, Encode, Decode, RuntimeDebug)]
+	pub struct RewardInfo<T: Config> {
+		pub total_reward: BalanceOf<T>,
+		pub last_paid: T::BlockNumber,
+	}
+
+	// No hooks
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Associate a native rewards_destination identity with a crowdloan contribution.
+		///
+		/// This is an unsigned call because the caller may not have any fnds to pay fees with.
+		/// This is inspired by Polkadot's claims pallet:
+		/// https://github.com/paritytech/polkadot/blob/master/runtime/common/src/claims.rs
+		///
+		/// This function and the entire concept of unassociated contributions may be obviated if
+		/// They will accept a memo filed in the Polkadot crowdloan pallet.
+		#[pallet::weight(0)]
+		pub fn associate_native_identity(origin: OriginFor<T>, reward_account: T::AccountId, proof: Vec<u8>) -> DispatchResultWithPostInfo {
+			ensure_none(origin)?;
+
+			//TODO check the proof:
+			// 1. Is signed by an actual unassociated contributor
+			// 2. Signs a valid native identity
+
+			//TODO update storage
+
+			//TODO Emit event
+
+			Ok(Default::default())
+		}
+
+		/// Collect whatever portion of your reward are currently vested.
+		#[pallet::weight(0)]
+		pub fn show_me_the_money(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			let payee = ensure_signed(origin)?;
+
+			// Calculate the veted amount on demand.
+			let mut info = AccountsPayable::<T>::get(payee).ok_or(Error::<T>::NoAssociatedClaim)?;
+			let now = frame_system::Module::<T>::block_number();
+
+			//TODO This part doesn't compile because of a million stupid errors about converting
+			// between u32, Balance, and BlockNumber. I think that is solvable, just annoying.
+			let payable_per_block = info.total_reward / T::VestingPeriod::get(); //TODO safe math;
+			let payable_period = T::VestingPeriod::get() - info.last_paid;
+			let payable_amount = payable_period * payable_per_block;
+
+			// Update the stored info
+			info.last_paid = now;
+			AccountsPayable::<T>::insert(&payee, &info);
+
+			// Make the payment
+			// TODO where are these reward funds coming from? Currently I'm just minting them right here.
+			// 1. We could have an associated type to absorb the imbalance.
+			// 2. We could have this pallet control a pot of funds, and initialize it at genesis.
```

Comment:
>  We could have this pallet control a pot of funds, and initialize it at genesis.

This makes sense. If the crowdloan -> moonbeam minting is only for a finite period of time and we know the total moonbeam that will be minted for this in total, we can fund an account for this pallet with the cap and distribute due funds by transferring from this pre-funded account.

Will this pot be capped? And is the crowdloan auction temporary (not continuous)?

I agree that even if we don't do the above, we need to track (and cap) total minted funds.